### PR TITLE
Bug 1753194 - Enable support for expiration by version on Android

### PR DESCRIPTION
**Pull request by Dexterp37**
This bumps the version of glean_parser currently being used by Glean and enables Android products to use expiration by version. To make sure things actually work, we change the Android sample to use expiration by version instead of a weird date.

### CODE REVIEW

**Comment by Dexterp37** [comment]
This is the error that shows up when using dates instead of versions in the registry files if expiration by version is turned on:

immagine

File: docs/user/language-bindings/android/android-build-configuration-options.md [hunk 1]
```diff
@@ -66,3 +66,15 @@ To override this, `ext.gleanYamlFiles` may be set to a list of explicit paths.
...
+
+Expire the metrics and pings by version, using the provided major version.
+
+If enabled, expiring metrics or pings by date will produce an error.
+
+```groovy
+ext.gleanExpireByVersion = 25
+```
+
+
```
Comment (badboy) [inline docs/user/language-bindings/android/android-build-configuration-options.md:80]:
```suggestion
```

too many newlines

File: samples/android/app/build.gradle [hunk 2]
```diff
@@ -17,6 +17,8 @@ android {
         applicationId "org.mozilla.samples.gleancore"
         minSdkVersion rootProject.ext.build['minSdkVersion']
         targetSdkVersion rootProject.ext.build['targetSdkVersion']
+        // Keep the version in sync with gleanExpireByVersion which is
+        // set below, near the end of the file.
         versionCode 1
```
Comment (badboy) [inline samples/android/app/build.gradle:22]: I hoped you pick something "fun" :(

File: samples/android/app/build.gradle [hunk 3]
```diff
@@ -57,6 +59,9 @@ dependencies {
 ext.gleanNamespace = "mozilla.telemetry.glean"
 // Fixed build date so we can test for it
 ext.gleanBuildDate = "2020-11-06T11:30:50+00:00"
+// Enable expiration by major version. The sample is at version 1,
+// see versionName.
+ext.gleanExpireByVersion = 1
```
Comment (badboy) [inline samples/android/app/build.gradle:64]: Will we have some guidance how this value should be fetched from build variables at build time?

File: samples/android/app/build.gradle (hunk 2 above)
Comment (Dexterp37) [reply samples/android/app/build.gradle:22]: Yeah, my day turned out to be "fun fun fun" :-P

File: samples/android/app/build.gradle (hunk 3 above)
Comment (badboy) [reply samples/android/app/build.gradle:64]: well, if we want Fenix to use that we probably have to figure out how they can set it, don't we?

File: build.gradle [hunk 4]
```diff
@@ -135,7 +135,7 @@ switch (DefaultPlatform.RESOURCE_PREFIX) {
         ext.nativeRustTarget = 'linux-x86-64'
         break
     case 'win32-x86-64':
-        ext.nativeRustTarget = 'win32-x86-64-gnu'
+        ext.nativeRustTarget = 'win32-x86-64-msvc'
```
Comment (badboy) [inline build.gradle:138]: this breaks CI

Comment (badboy) [reply build.gradle:138]: Let's revert that and maybe open a bug so we can handle that afterwards?

Comment (Dexterp37) [reply build.gradle:138]: Eugh, sorry, this was not intentional. Let me revert changes to this file.

**Pull request merged**